.github/workflows/increment_guard.yml

```diff
@@ -0,0 +1,26 @@
+# Ensures that the current lib version is not yet published but executing the Gradle
+# `checkVersionIncrement` task.
+
+name: Check version increment
+
+on:
+  push:
+    branches: '**'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v1
+
+      - uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
```

Comment:
I guess the v2 suggestions apply to all other CI configs, so I won't duplicate myself.